pkg/runit/test_helper.go

```diff
@@ -20,24 +22,56 @@ func FakeChpst() string {
 	return util.From(runtime.Caller(0)).ExpandPath("fake_chpst")
 }
 
-func FakeServiceBuilder() *ServiceBuilder {
-	testDir := os.TempDir()
-	fakeSBBinPath := util.From(runtime.Caller(0)).ExpandPath("fake_servicebuilder")
-	configRoot := filepath.Join(testDir, "/etc/servicebuilder.d")
-	os.MkdirAll(configRoot, 0755)
-	_, err := os.Stat(configRoot)
+// testServiceBuilder is a ServiceBuilder for use in unit tests.
+type testServiceBuilder struct {
+	root string
+	ServiceBuilder
+}
+
+// Cleanup removes the file system changes made by the testServiceBuilder.
+func (s testServiceBuilder) Cleanup() {
+	os.RemoveAll(s.root)
+}
+
+// mustMkdirAll creates the given directory or dies trying
+func mustMkdirAll(path string) {
+	if err := os.MkdirAll(path, 0755); err != nil {
+		panic(err)
+	}
+}
+
+// FakeServiceBuilder constructs a testServiceBuilder for use in unit tests. It is the
+// caller's responsibility to always call Cleanup() on the return value to ensure that
+// file system changes are removed when this test ends.
+func FakeServiceBuilder() (s *testServiceBuilder) {
+	root, err := ioutil.TempDir("", "runit_test")
 	if err != nil {
-		panic("unable to create test dir")
+		panic(err)
 	}
-	stagingRoot := filepath.Join(testDir, "/var/service-stage")
-	os.MkdirAll(stagingRoot, 0755)
-	runitRoot := filepath.Join(testDir, "/var/service")
-	os.MkdirAll(runitRoot, 0755)
-
-	return &ServiceBuilder{
-		ConfigRoot:  configRoot,
-		StagingRoot: stagingRoot,
-		RunitRoot:   runitRoot,
-		Bin:         fakeSBBinPath,
+	fmt.Printf("created test root: %s\n", root)
```

Comment:
I'm OK with it in (it's only visible if the test fails), but I'm also OK removing it.